Source/WPEFramework/PluginServer.h

```diff
@@ -1290,32 +1308,46 @@ namespace PluginHost {
                     : Core::JSON::Container()
                     , AutoStart()
                     , Configuration(_T("{}"), false)
+                    , SystemRootPath()
+                    , Startup()
                 {
                     Add(_T("autostart"), &AutoStart);
                     Add(_T("configuration"), &Configuration);
+                    Add(_T("systemrootpath"), &SystemRootPath);
+                    Add(_T("startup"), &Startup);
                 }
-                Plugin(const string& config, const bool autoStart)
+                Plugin(const string& config, const bool autoStart, const string& systemRootPath, const PluginHost::IShell::startup value)
                     : Core::JSON::Container()
                     , AutoStart(autoStart)
                     , Configuration(config, false)
+                    , SystemRootPath(systemRootPath)
+                    , Startup(value)
                 {
                     Add(_T("autostart"), &AutoStart);
                     Add(_T("configuration"), &Configuration);
+                    Add(_T("systemrootpath"), &SystemRootPath);
+                    Add(_T("startup"), &Startup);
                 }
                 Plugin(Plugin const& copy)
                     : Core::JSON::Container()
                     , AutoStart(copy.AutoStart)
                     , Configuration(copy.Configuration)
+                    , SystemRootPath(copy.SystemRootPath)
+                    , Startup(copy.Startup)
                 {
                     Add(_T("autostart"), &AutoStart);
                     Add(_T("configuration"), &Configuration);
+                    Add(_T("systemrootpath"), &SystemRootPath);
+                    Add(_T("startup"), &Startup);
                 }
 
                 ~Plugin() override = default;
 
             public:
                 Core::JSON::Boolean AutoStart;
                 Core::JSON::String Configuration;
+                Core::JSON::String SystemRootPath;
+                Core::JSON::EnumType<PluginHost::IShell::startup> Startup;
```

Comment:
By adding this value, I guess we can drop the Autpostart boolean.

Reply:
done